### PR TITLE
Added Market resource, package name update, code clean up

### PR DESCRIPTION
**Pull request by autumnfound**
Change-Id: Ib8d095f06d59209ed894a1048a4ca8415b81c59a
Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

### CODE REVIEW

File: src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * which is available at http://www.eclipse.org/legal/epl-v20.html,
  * SPDX-License-Identifier: EPL-2.0
  */
-package org.eclipsefoundation.marketplace.config;
+package org.eclipse.foundation.marketplace.config;
```
Comment (mbarbero) [inline src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java]: actually, I liked better the eclipsefoundation (single word) package name. Why did you change that?

File: src/main/java/org/eclipse/foundation/marketplace/dto/Category.java [hunk 2]
```diff
@@ -69,57 +65,49 @@ public void setUrl(String url) {
...
+		result = prime * result + ((id == null) ? 0 : id.hashCode());
 		result = prime * result + ((name == null) ? 0 : name.hashCode());
 		result = prime * result + ((url == null) ? 0 : url.hashCode());
 		return result;
 	}
 
 	@Override
 	public boolean equals(Object obj) {
-		if (this == obj)
+		// check basic object properties
```
Comment (mbarbero) [inline src/main/java/org/eclipse/foundation/marketplace/dto/Category.java]: Don't code equals by yourself, use Objects.equals https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-

File: src/main/java/org/eclipse/foundation/marketplace/dto/Category.java [hunk 3]
```diff
@@ -69,57 +65,49 @@ public void setUrl(String url) {
...
-
-	/**
-	 * @param marketIds the marketIds to set
-	 */
-	public void setMarketIds(List<Integer> marketIds) {
-		this.marketIds = new ArrayList<>(marketIds);
-	}
-
 	@Override
 	public int hashCode() {
```
Comment (mbarbero) [inline src/main/java/org/eclipse/foundation/marketplace/dto/Category.java]: Don't code hashcode by yourself, use Object.hash(Object...) https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...-

File: src/main/java/org/eclipse/foundation/marketplace/dto/Listing.java [hunk 4]
```diff
@@ -411,11 +395,186 @@ public void setVersions(List<SolutionVersion> versions) {
 		this.versions = new ArrayList<>(versions);
 	}
 	
+	
+	
+	@Override
+	public int hashCode() {
```
Comment (mbarbero) [inline src/main/java/org/eclipse/foundation/marketplace/dto/Listing.java]: Sic

File: src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java (hunk 1 above)
Comment (autumnfound) [reply src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java]: I matched the package naming found within the ECA plugin, but I can see now that the bot api that you created has single word...we should choose which we are moving forward with for future APIs. I agree that one word looks nicer. Should I revert that change?

File: src/main/java/org/eclipse/foundation/marketplace/dto/Category.java (hunk 2 above)
Comment (autumnfound) [reply src/main/java/org/eclipse/foundation/marketplace/dto/Category.java]: I used the auto-generated equals from Eclipse. I think there was an option to use Object.equals instead. I'll try that instead.

File: src/main/java/org/eclipse/foundation/marketplace/dto/Category.java (hunk 3 above)
Comment (autumnfound) [reply src/main/java/org/eclipse/foundation/marketplace/dto/Category.java]: Same as the object.equals stuff, auto-generated. I'll switch over.

File: src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java (hunk 1 above)
Comment (mbarbero) [reply src/main/java/org/eclipse/foundation/marketplace/config/JsonBConfig.java]: I did not create the ECA plugin, so I've no history about the choosing of package name. I think that we should stick to "eclipsefoundation" for all of our future work. If it's not too much burden, yes please revert the package rename. Thanks.
